airbyte-cdk/python/airbyte_cdk/sources/file_based/schema_helpers.py

```diff
@@ -74,7 +112,30 @@ def conforms_to_schema(record: Mapping[str, Any], schema: Mapping[str, str]) ->
     - For every column in the record, that column's type is equal to or narrower than the same column's
       type in the schema.
     """
-    ...
+    schema_columns = set(schema.get("properties", {}).keys())
+    record_columns = set(record.keys())
+
+    if not record_columns.issubset(schema_columns):
+        return False
+
+    for column, definition in schema.get("properties", {}).items():
+        expected_types = definition.get("type", [])
+        value = record.get(column)
+
+        if not isinstance(expected_types, list):
+            expected_types = [expected_types]
+
+        if value is not None:
+            if "object" in expected_types:
+                if not isinstance(value, dict):
+                    return False
+            elif "array" in expected_types:
+                if not isinstance(value, list):
```

Comment:
While testing I realized that this one is a little more complicated since we want to check that the types in the list conform to the schema, so I updated to do so.